### ISSUE TITLE
react-select mistakes an object as "not in the list" after just adding it.

### ISSUE THREAD

**Issue by StokeMasterJack** [JedWatson/react-select#1346]
I am using an async creatable. 

Use-case:
1. User enters a value, say "Pillars of the Earth", that is not in the list.
2. User presses Enter key, thus selecting the [Create option "Pillars of the Earth"] item.
3. In the onChange event handler I catch this item and detect that it has className "Select-create-option-placeholder". So I add it to the database (using async call to server).
4. When async call returns from server, I update react-select's value prop.
5. This succeeds and thus the new value, "Pillars of the Earth" is now displayed as react-select's value.

So here is the problem. react-select still considers "Pillars of the Earth" to _not_ be in the list. And thus it still shows [Create option "Pillars of the Earth"] as the first list item.
Thus, when my user presses tab, to go to the next field, the tab key is treated like enter and thus selects the first item in the list. This repeats the whole process over again, creating another "Pillars of the Earth"

I suspect the reason it does this is that it's not updating the options list. And so it still thinks that this item is not in the list.

In short, I need react-select to treat the newly added "Pillars of the Earth" as "in the list" so that it will not prompt the user to add it again.

**Comment by akshayvenugopal** [comment]
Facing the same issue here I have debugged the code and cannot find the issue what is it that goes wrong

**Comment by rossfishkind** [comment]
@StokeMasterJack did you find a workaround/fix to this? Thanks!

**Comment by bladey** [comment]
Hello -

In an effort to sustain the `react-select` project going forward, we're closing old issues / pull requests.

We understand this might be inconvenient but in the best interest of supporting the broader community we have to direct our limited efforts to maintain the latest version.

If you feel this issue / pull request is still relevant and you'd like us to review it, please leave a comment and we'll do our best to get back to you.
